os/arch/arm/src/amebasmart/amebasmart_reboot_reason.c

```diff
@@ -112,6 +118,16 @@ static reboot_reason_code_t up_reboot_reason_get_hw_value(void)
 		}
 		/* KM0: IWDG reset*/
 		else if (boot_reason & AON_BIT_RSTF_IWDG) {
+			/* KM0: IWDG is used in KM0 to ensure KM0 is working properly, KM4 is wakeup properly from PG, CA32 power is power-on from PG
+			* backup register will be used to record the case that triggered IWDG reboot
+			*/
+			if (boot_reason_reg2 == 0x2) {
```

Comment:
Is this right to compare with the value? not bit masking?

Reply:
When designing amebasmart's reboot reason, I made reference to amebadD for up_reboot_reason_write which writes the reboot reason code directly into BKUP register instead of bit wise operation.
So for amebasmart's km4 fault or secure fault, it followed the same implementation. As i checked in reboot_reason.h, there is no multiple reboot reason code.